kombu/utils/url.py

```diff
@@ -30,8 +30,7 @@ class urlparts(NamedTuple):
     query: Mapping
 
 
-def parse_url(url):
-    # type: (str) -> Dict
+def parse_url(url: str) -> Dict[str, Any]:
```

Comment:
Unfortunately, `TypedDict` doesn't support extra keys, so this is the best we can get... More information on: https://github.com/python/mypy/issues/4617

Reply:
anything but Any

Reply:
I'm still unsure about `Dict[str, Union[str, int, Dict[str, str], 'ssl.VerifyMode']]`. If someone could double-check that these are the right types in that union it would be great.

Reply:
On my side, I think `Dict[str, Any]` is the right call.

Another possibility would be (which I'm not sure...):
```python
class ParsedURL(TypedDict):
    transport: str
    hostname: str
    port: int
    userid: str
    password: str
    virtual_host: str

def parse_url(url: str) -> Union[ParsedURL, Dict[str, Any]]
```

Reply:
Other than that, you can check if you can create the exact type with some information provided on this issue: https://github.com/python/mypy/issues/4617